File: tests/login_test.js
const { Builder, By, until } = require('selenium-webdriver');

async function testLogin() {
  // Initialize the Chrome browser
  console.log("Starting browser...");
  let driver = await new Builder().forBrowser('chrome').build();

  try {
    // Navigate to the login page
    console.log("Navigating to login page...");
    await driver.get('http://localhost:3000/login');
    await driver.manage().window().maximize();

    // Wait until the email field is rendered
    await driver.wait(until.elementLocated(By.id('userEmail')), 10000);

    const emailField = await driver.findElement(By.id('userEmail'));
    const passwordField = await driver.findElement(By.id('userPass'));
    const submitBtn = await driver.findElement(By.css("button[type='submit']"));

    // Enter credentials
    const testEmail = "test@05gmail.com";
    const testPassword = "Test@05";

    console.log(`Entering Email: ${testEmail}`);
    await emailField.sendKeys(testEmail);

    console.log("Entering Password...");
    await passwordField.sendKeys(testPassword);

    // Optional: Test the eye toggle
    try {
      const eyeButton = await driver.findElement(By.css("button[aria-label='Show password']"));
      await eyeButton.click();
      await driver.sleep(1000); // Wait 1 second to visibly see the text
    } catch(err) {
        // ignore if button not found
    }

    console.log("Clicking Submit...");
    await driver.executeScript("arguments[0].click();", submitBtn);

    // Verify Result: Wait to see if we get routed to the dashboard URL
    await driver.wait(until.urlContains('/dashboard'), 5000);
    
    console.log("\n✅ TEST PASSED: Successfully logged in and redirected to the Dashboard!");

  } catch (error) {
    console.log(`\n❌ TEST FAILED: ${error.message}`);
  } finally {
    // Keep browser open for a short time to review, then quit
    await driver.sleep(3000);
    console.log("Closing browser...");
    await driver.quit();
  }
}

testLogin();

File: tests/login_test.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

def test_login():
    # Setup Chrome WebDriver
    print("Starting browser...")
    driver = webdriver.Chrome()
    
    try:
        # Navigate to the Login Page
        print("Navigating to login page...")
        driver.get("http://localhost:3000/login")
        driver.maximize_window()
        
        # Wait for the email input field to load (id="userEmail")
        wait = WebDriverWait(driver, 10)
        email_field = wait.until(EC.presence_of_element_located((By.ID, "userEmail")))
        password_field = driver.find_element(By.ID, "userPass")
        submit_button = driver.find_element(By.CSS_SELECTOR, "button[type='submit']")
        
        # Action: Enter credentials
        test_email = "test@05gmail.com"
        test_password = "Test@05"
        
        print(f"Entering Email: {test_email}")
        email_field.send_keys(test_email)
        
        print("Entering Password...")
        password_field.send_keys(test_password)
        
        # Optional: Test the eye toggle!
        try:
            eye_button = driver.find_element(By.CSS_SELECTOR, "button[aria-label='Show password']")
            eye_button.click()
            time.sleep(1) # Pause to visibly see the password is typed correctly
        except Exception as e:
            pass
            
        print("Clicking Submit...")
        driver.execute_script("arguments[0].click();", submit_button)
        
        # Verify Results: Wait to see if we get routed to the dashboard
        # Assuming the dashboard URL is http://localhost:5173/dashboard
        wait.until(EC.url_contains("/dashboard"))
        print("\n✅ TEST PASSED: Successfully logged in and redirected to the Dashboard!")
        
    except Exception as e:
        print(f"\n❌ TEST FAILED: {str(e)}")
        
    finally:
        # Pause so you can see the result before closing
        time.sleep(3)
        print("Closing browser...")
        driver.quit()

if __name__ == "__main__":
    test_login()
